src/controllers/DegreeController.ts

```diff
@@ -0,0 +1,73 @@
+import { Request, Response} from 'express'
+import Degree from '../models/Degree'
+
+class DegreeController{
+    static async getAllDegrees(req: Request, res: Response){
+        try{
+            const degrees = await Degree.find().populate('courses').populate('students')
+            res.json(degrees)
+        }catch(error){
+            console.log(error)
```

Comment:
please replace all console.log with log4js library (replace it with logger.info('...'), logger.error('....'))